### PR TITLE
Extract date of last modification from git log

### PR DESCRIPTION
**Pull request by hagenw**
As @mgeier proposed a few month ago I added the last modification date to the footer.

The modification date is extracted from the last git commit. The extracted year is then also used to set the copyright year. The actual date is added to the footer. This requires more work as we have to use our custom `footer.html` template and pass the variable to the template via `html_context`.

The result can be inspected at http://sfstoolbox.org/en/v3-date/.

### CODE REVIEW

**Comment by mgeier** [comment]
I think this is much too complicated for the questionable benefit it creates.

We might as well just include the date in the auto-generated version number and put this somewhere (I guess this would work with less effort).

But either way, the "modified date" information is actually wrong, since it displays the last change of the Git repository. I think if anything, it would make more sense to show the last-changed-date for each page individually. But I guess this would be much more complicated.

File: conf.py [hunk 1]
```diff
@@ -16,8 +16,15 @@
 
 # -- GENERAL -------------------------------------------------------------
 
+def command_output(cmd):
+    output = subprocess.check_output(cmd.split(' '))
```
Comment (mgeier) [inline conf.py:20]: I think it makes more sense to hand-craft the argument lists here, but if you have to use strings, you should use https://docs.python.org/3/library/shlex.html#shlex.split to split them.

File: conf.py [hunk 2]
```diff
@@ -16,8 +16,15 @@
...
+    return output.decode().strip()
+
+date_modified = command_output('git log -1 --date=short --format=%cd')
```
Comment (mgeier) [inline conf.py:23]: This breaks if `git` is not found or if called outside of a Git repo (e.g. when using a tarball).

File: conf.py (hunk 1 above)
Comment (hagenw) [reply conf.py:20]: I guess you refer to shlex.split for command parsing due to problems like this with split: https://github.com/python-telegram-bot/python-telegram-bot/issues/568

**Comment by hagenw** [comment]
Good point that the current implementation is meaningless for multiple pages.
I agree that it would be too much effort to make it work for single pages and will just close this request.

File: conf.py (hunk 1 above)
Comment (mgeier) [reply conf.py:20]: Yes. I wouldn't really expect those problems here, but out of principle I think it's better to use the tool that was created for exactly that use case.

**Pull request closed without merging**